File: 605.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int s = flowerbed.size();
        if(s == 1) return (flowerbed[0] + n) <= 1;
        for(int i=0; i<s; i++){
            if(!n) break;
            if(i == 0){
                if(!flowerbed[0] && !flowerbed[1]){
                    flowerbed[0]++;
                    n--;
                    i++;
                }
            }
            else if(i == s-1){
                if(!flowerbed[s-1] && !flowerbed[s-2]) n--;
            }
            else if(!flowerbed[i-1] && !flowerbed[i] && !flowerbed[i+1]){
                flowerbed[i]++;
                n--;
                i++;
            }
        }
        return !n;
    }
};

int main(){
    Solution s;
    int flowerNum;
    cin >> flowerNum;
    vector<int> flowerbed(flowerNum);
    for(int i=0; i<flowerNum; i++) cin >> flowerbed[i];
    int n;
    cin >> n;
    cout << s.canPlaceFlowers(flowerbed, n) << endl;
}

File: 605.py
class Solution:
    def canPlaceFlowers(self, flowerbed: list[int], n: int) -> bool:
        s = len(flowerbed)
        if s == 1:
            return (flowerbed[0] + n) <= 1
        for i in range(s):
            if n == 0:
                break
            if i == 0:
                if flowerbed[0] == 0 and flowerbed[1] == 0:
                    flowerbed[0] += 1
                    n -= 1
                    i += 1
            elif i == s-1:
                if flowerbed[s-1] == 0 and flowerbed[s-2] == 0:
                    n -= 1
            elif flowerbed[i-1] == 0 and flowerbed[i] == 0 and flowerbed[i+1] == 0:
                flowerbed[i] += 1
                n -= 1
                i += 1
        return n == 0
    
if __name__ == '__main__':
    s = Solution();
    flowerbed = list(map(int, input().split()))
    n = int(input());
    print(s.canPlaceFlowers(flowerbed, n))
    